Makefile

```diff
@@ -149,6 +216,40 @@ tools: $(CONTROLLER_GEN) \
 lint: $(GOLANGCI_LINT)
 	$(GOLANGCI_LINT) run ./... --fix
 
+.PHONY: jsonnet-fmt
+jsonnet-fmt: $(JSONNETFMT)
+	find jsonnet/ -name 'vendor' -prune \
+	  -o -name '*.libsonnet' -print \
+		-o -name '*.jsonnet' -print \
+	| xargs -n 1 -- $(JSONNETFMT) $(JSONNETFMT_ARGS) -i
+
+.PHONY: jsonnet-lint
+jsonnet-lint: $(JSONNETLINT) jsonnet-vendor
+	find jsonnet/ -name 'vendor' -prune \
+	  -o -name '*.libsonnet' -print \
+		-o -name '*.jsonnet' -print \
+	| xargs -n 1 -- $(JSONNETLINT) -J $(JSONNET_VENDOR)
+
+.PHONY: check-jq
+check-jq:
+	jq --version > /dev/null
+
+jsonnet-vendor: $(JSONNET_BUNDLER)
+	rm -rf jsonnet/vendor
+	cd jsonnet && $(JSONNET_BUNDLER) install
+
+.PHONY: generate-prometheus-rules
+generate-prometheus-rules: jsonnet-tools check-jq jsonnet-vendor kustomize
+	for dir in jsonnet/components/*/; do \
+		component=$$(basename $$dir) ;\
+		echo "Generating prometheusrule file for $$component" ;\
+		$(JSONNET) -J $(JSONNET_VENDOR) $$dir/main.jsonnet \
+		| jq .rule \
```

Comment:
How do we ensure that this target is called? 
Should this be added to `generate` target?